response_operations_ui/common/redis_cache.py

```diff
@@ -0,0 +1,75 @@
+import json
+import logging
+
+from flask import current_app
+from redis.exceptions import RedisError
+from structlog import wrap_logger
+
+from response_operations_ui import redis
+from response_operations_ui.controllers.cir_controller import get_cir_metadata
+from response_operations_ui.controllers.survey_controllers import (
+    get_survey_by_shortname,
+)
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+class RedisCache:
+    SURVEY_EXPIRY = 600  # 10 mins
+
+    def get_cir_metadata(self, survey_ref, formtype):
+        """
+        Gets the cir_metadata from redis or the cir service
+
+        :param short_name: str: the qualifying part of the redis key
+                                (response-operations-ui:survey:<SURVEY_REF>:<FORMTYPE>)
+        :return: Result from either the cache or the CIR service
+        """
+        redis_key = f"response-operations-ui:cir:{survey_ref}:{formtype}"
+        try:
+            result = current_app.redis.get(redis_key)
+        except RedisError:
+            logger.error("Error getting value from cache", key=redis_key, exc_info=True)
+            result = None
+
+        if not result:
+            logger.info("Key not in cache, getting value from CIR service", key=redis_key)
+            result = get_cir_metadata(survey_ref, formtype)
+            current_app.redis.set(redis_key, json.dumps(result), self.SURVEY_EXPIRY)
+            return result
+
+        return json.loads(result.decode("utf-8"))
+
+    def get_survey_by_shortname(self, short_name):
```

Comment:
Same as above, the function isn't name quite correct, considering there is a set in here. Realistically these should be separated

Reply:
As above, I think this is fine. The set is not referring to the underlying data object being updated (i.e. in the db) it's the caching part of the flow. The method name (rightly) mirrors the existing/original non-cache method it's implementing the cache for